### PR TITLE
Update vNext FluentUI Button to support pointer interaction

### PR DESCRIPTION
**Pull request by huwilkes**
### Platforms Impacted
- [X] iOS
- [ ] macOS

### Description of changes

Added pointer support to the button when the button is enabled. After talking to design, we decided to go with the system hoverEffect rather than implementing a custom .onHover.

### Verification
Tested on simulator

| Light                                       | Dark                                      |
|----------------------------------------------|--------------------------------------------|
| PrimaryPointerSupport_Light | PrimaryPointerSupport_Dark |
| SecondaryPointerSupport_Light | SecondaryPointerSupport_Dark |
| GhostPointerSupport_Light | GhostPointerSupport_Dark |
| AccentFABPointerSupport_Light | AccentFABPointerSupport_Dark |
| SubtleFABPointerSupport_Light | SubtleFABPointerSupport_Dark |

### Pull request checklist

This PR has considered:
- [X] Light and Dark appearances
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/737)

### CODE REVIEW

File: ios/FluentUI/Vnext/Button/Button.swift [hunk 1]
```diff
@@ -126,55 +126,120 @@ struct FluentButtonBody: View {
...
-                    .scalableFont(font: tokens.textFont,
-                                  shouldScale: !isFloatingStyle)
-                    .modifyIf(isFloatingStyle, { view in
-                        view.frame(minHeight: tokens.textMinimumHeight)
-                    })
-                }
+        @ViewBuilder
+        var buttonContent: some View {
+            HStack(spacing: tokens.interspace) {
+                        if let image = state.image {
```
Comment (mischreiber) [inline ios/FluentUI/Vnext/Button/Button.swift:153]: A lot of the spacing is off in this file. Could you do a `^I` on the whole file to align it to Apple standard?

File: ios/FluentUI/Vnext/Button/Button.swift [hunk 2]
```diff
@@ -126,55 +126,120 @@ struct FluentButtonBody: View {
...
+            shadow2DepthY = tokens.restShadow2DepthY
+        }
+
+        @ViewBuilder
+        var button: some View {
+            if isFloatingStyle {
+                buttonBackground
+                    .clipShape(Capsule())
+                    .shadow(color: shadow1Color,
+                           radius: shadow1Blur,
```
Comment (mischreiber) [inline ios/FluentUI/Vnext/Button/Button.swift:227]: Note: we don't align colons (`:`) in Swift like we do in Objective-C. The aforementioned `^I` will left-align these as is the Swift standard.

**Review by mischreiber — COMMENTED** [review]
Great cleanup! This is super readable now.

Is there anything we can do about the shadow disappearing on our Floating Style buttons? And if not, has design given a thumbs-up to ship as is?

**Comment by huwilkes** [comment]
> Great cleanup! This is super readable now.
> 
> Is there anything we can do about the shadow disappearing on our Floating Style buttons? And if not, has design given a thumbs-up to ship as is?

I haven't been able to find a way to keep the shadow around with a hover effect. I've reached out to Eric about if he thinks it's fine as is, but he hasn't replied yet. When I showed it to him earlier, though, he only mentioned that the shape of the cursor should match the shape of the FAB better. But, I'll update this once I get a reply from him.

File: ios/FluentUI/Vnext/Button/Button.swift [hunk 3]
```diff
@@ -126,55 +126,120 @@ struct FluentButtonBody: View {
...
+            })
+            .frame(maxWidth: .infinity, maxHeight: .infinity)
+            .foregroundColor(Color(titleColor))
+        }
+
+        @ViewBuilder
+        var buttonBackground: some View {
+            if tokens.borderSize > 0 {
+                buttonContent.background(
+                    AnyView(RoundedRectangle(cornerRadius: tokens.borderRadius)
```
Comment (rdeassis) [inline ios/FluentUI/Vnext/Button/Button.swift:182]: You don't need the expensive AnyView() wrapping anymore on the RoundedRectangles in the if/else cases.

**Pull request merged**